### ISSUE TITLE
monte carlo: get_lat_lon

### ISSUE THREAD

**Issue by RachaelDMueller** [MIDOSS/MIDOSS-MOHID-config#9]
General Monte Carlo overview plan is[ here](https://docs.google.com/document/d/1MRDCViRZelm_RPVUDd67SdIufYHI4-KdcpwK9zCXxnY/edit)

Function: get_lat_lon
Input: spill month (mm)
Select: Use spill month designation to select 1 of the 12 Geotiffs representing all vessel traffic by month (Naming convention: all_traffic_01.tif, all_traffic_02,.tif all_traffic_03.tif...).  Decide: Project raster in geographic before saving to geotiff so that x/y coordinates are in lat/lon 
R2: Apply Vicky’s random selection algorithm to select a lat/lon location
The value determined will be the designated lat/lon of the 1km x 1km raster grid.  We need to determine whether this lat/lon is center or corner of grid.  We will apply another random selection toward next to refine lat/lon spill location to a 4 decimal place precision that queries the salish sea mask to verify as water point. 
Output: geotiff_box_lat [1x1 km resolution], geotiff_box_lon [1x1 km resolution], geotiff_box_lat, geotiff_box_lon

**Comment by RachaelDMueller** [comment]
First draft is [here](https://github.com/MIDOSS/analysis-rachael/blob/master/notebooks/monte_carlo/get_lat_lon.ipynb).  Issues: 

- This script is tailored to our sample “September" geotiff. Need to make more general (look for DateTime simplification!)
- The probability is based on traffic for the individual month.  I mention here in case we decide that we want the probability to be based on annual vessel time exposure rather than monthly  vessel time exposure.

**Comment by RachaelDMueller** [comment]
The method that we are using has always appeared downsampled to me.  I explored this possibility a bit more today but don’t see an obvious reason why this subsampling would occur.  The raster says it’s spacing is 1, and we use an inherent spacing of 1 with the range() function...so the resolutions ought to match.  It just doesn’t match to my eye. 

As I considered today the task of sub-sampling our lat/lon value in the 1km x 1km geotiff grid, it occured to me that the transform function can operate to either down or upsample the raster image.  Why not kill two birds with one stone? 

I modified [the script](https://github.com/MIDOSS/analysis-rachael/blob/master/notebooks/monte_carlo/get_lat_lon.ipynb) in an inelegant but effective way with a subsample variable.

The problem that I ran into is that the latitude and longitude pair array needs to be the same size as the probability array; so I then tried to subsample the geotiff using a method that I read about [here](https://rasterio.readthedocs.io/en/latest/topics/resampling.html).  Unfortunately, the result doesn't appear to work. When I searched the problem, I found that other users have had problems with rosario for down/up sampling as well and another tool is recommended.  

My plan is to leave the method “as is” for now, as I/we have other priorities.  We can always revisit.
